docs/NFF-0.1-spec.norg

```diff
@@ -0,0 +1,875 @@
+     segment. However, creating a level 1 ordered list segment inside would be okay.+
+
+    @code norg
+    - \
+      The indent segment extension has started.
+
+      I can now include multiple paragraphs within this list item.
+
+      @code lua
+      print("Even code blocks are allowed!")
+      @end
+
+      ^ Range-able detached modifiers can be inserted, too.
+
+      ---
+    - Another nestable detached modifier ends the indent segment
+    - However, in this example I already closed the indent segment manually with `---`
+    @end
+
+* Tags
+  Tags are _Norg_'s official way of extending its own syntax.
+
+  There are 3 different tag types, each with their own way of changing the way text in _Norg_ is
+  interpreted. Before we discuss those, however, we should discuss the syntax rules for tags:
+  - A tag is similar to a {# detached modifiers}[detached modifier] in the sense that it must begin
+    at the beginning of a line with optional whitespace (but nothing else) preceding it.
+  - After that you will encounter the special tag character (`|`, `@` or `#`), /none/ of which are
+    attached modifiers - if those chars were attached modifiers then we would have ambiguity on our
+    hands. The special tag character is then /immediately/ followed by text, which becomes the /tag
+    name/. Said tag name can consist of any {# alphanumeric characters}[alphanumeric character]
+    and/or `-`, `_` and `.`.
+  - Tags can have their names delimited by a `.` in order to create a "hierarchy", e.g.
+    `document.meta`.
+  - \
+    After a whitespace character any number of parameters on the same line may follow:
+    @code norg
+    |tag-name.subtag parameter1 parameter2
+    @end
+
+    Parameters may consist of any character (apart from a {# line break}, of course).
+    ---
+
+   _Norg_ provides several inbuilt tag names that are reserved, but their details are not explained
+   in this specification - this document strictly covers syntax. There is no restriction in regards
+   to the length of a tag name, nor are there any disallowed names that a parser should omit
+   (unless they don't adhere to the above rules regarding tag names).
+
+** Carryover Tags
+   A carryover tag is a way to assign some data to the next element in the document.
+   +TODO: Finish+
+   @code norg
+   |carryover-tag param1 param2
+   @end
+
+** Ranged Tags
+   Ranged tags are a way to express custom information within a range.
+   They begin with the traditional tag declaration and are ended with an `end` statement.
+   They allow you to create your own completely custom syntax within a determined boundary.
+
+   The `end` statement has a simple ruleset:
+   - Must be at the start of a line, may be preceded by any whitespace (but nothing else)
+   - Must use the same prefix as its parent (in the case of regular ranged tags: `#`; in the case
+     of verbatim tags: `@`)
+   - Must not be succeeded by anything other than a {# line break}.
+
+*** Ranged Tags
```

Comment:
```suggestion
*** Standard Ranged Tags
```
Just a suggestion to de-duplicate the heading

Reply:
I just saw that below you talk about "regular" rather than "standard" ranged tags. In the top of the document we talked about "standard detached mods". Shall we stick to one phrasing?